### PR TITLE
Feature/pre

### LINKED ISSUE #56: Feature/Pre
Pre formatted text

### PR DESCRIPTION
**Pull request by Snugug**
Resolves #56

### CODE REVIEW

**Comment by Snugug** [comment]
Doesn't include line numbers, as that requires JavaScript and we'll be using Prism for enhancement with through a component; different PR

File: patterns/base/pre/sass/partials/_extends.scss [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  word-spacing: normal;
+  text-align: left;
+  word-break: normal;
+  white-space: pre;
+  direction: ltr;
+  tab-size: 2;
+}
+
+%base--pre {
+  @extend %base--pre-build;
```
Comment (una) [inline patterns/base/pre/sass/partials/_extends.scss]: Why is this separated? When will be use pre-build outside of this?

Comment (Snugug) [inline patterns/base/pre/sass/partials/_extends.scss]: Because it's used for nested code as well

Comment (Snugug) [inline patterns/base/pre/sass/partials/_extends.scss]: https://github.com/Snugug/patterns/commit/1a757f6f030989768d35252c75ee3a09860e2a70#diff-2872dd0fc8fca5a7976907954762423cR22

Comment (una) [inline patterns/base/pre/sass/partials/_extends.scss]: :+1:

File: patterns/base/pre/sass/partials/_extends.scss [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  margin: .5em 0;
+  overflow: auto;
+
+  border: 1px solid rgba(color('gray'), .25);
+
+  code,
+  .base--code {
+    @extend %base--pre-build;
+    padding: 0;
+    font-size: 1em;
```
Comment (una) [inline patterns/base/pre/sass/partials/_extends.scss]: Do you need to set this explicitly?

Comment (Snugug) [inline patterns/base/pre/sass/partials/_extends.scss]: Wadaya mean set explicitly?

We need to update how code works when nested inside `pre`, which is different than just inline `code`

**Pull request merged**